### PR TITLE
Add more checks to emitc::CallOp verifier

### PR DESCRIPTION
**Pull request by OliverScherf**
This adds the following checks to the `emitC::CallOp` verifier:
- args: arguments need a type
- template_args: arguments need a type, forbid float literals, forbid dense elements attributes

### CODE REVIEW

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:66]:
```suggestion
      // Args with elements of type index must be in range [0..operands.size).
```

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:76]:
```suggestion
     // Args with elements of type ArrayAttr must have a type.
```

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:92]: I think if rephrased, those sentences can start with a capital letter.

**Review by marbre — CHANGES_REQUESTED** [review]
Will take a closer look tomorrow.

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (OliverScherf) [reply lib/Dialect/EmitC/IR/EmitCDialect.cpp:92]: I rephrased it, please take a look at it

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:92]:
```suggestion
      // Template args with elements of type ArrayAttr are not allowed
```

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:66]: Maybe this comment should be moved down to the inner if-clause?

**Review by marbre — CHANGES_REQUESTED** [review]
Only minor nits.

File: lib/Dialect/EmitC/IR/EmitCDialect.cpp
Comment (marbre) [inline lib/Dialect/EmitC/IR/EmitCDialect.cpp:93]:
```suggestion
      // Template args with elements of type ArrayAttr are not allowed.
```

One more, didn't noticed it before :see_no_evil:

**Review by marbre — COMMENTED** [review]
Looks good to me, but let's wait for Simon's approval.

**Review by marbre — COMMENTED** [review]
(Needed to re-review to get rid of the changes requested flag... which obviously does not work. Will approve after Simon did.)

**Review by simon-camp — APPROVED** [review]
LGTM :)

**Review by marbre — APPROVED** [review]

**Comment by marbre** [comment]
I messed up during approving changes and re-requested reviews, but @simon-camp as well as I actually approved this.

**Pull request merged**
